.github/workflows/test.yaml

```diff
@@ -0,0 +1,34 @@
+name: Test
+
+on: [push]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install uv
+        uses: astral-sh/setup-uv@v5
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version-file: "pyproject.toml"
+      - name: Install the project
+        run: uv sync --dev
+      - name: Run tests
+        run: uv run pytest
+  docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Build and push Docker image
```

Comment:
Where does this push to, and using which tags? I would not want a new `:latest` for every commit on main ...

Reply:
As you can see in line 34, this job doesn't push the image.

We'd need to add another workflow that only runs on tags to automatically build and push the image to e.g ghcr or docker hub.

There's two reasons why we're building the image here:

1. To ensure we don't accidentally merge commits that break the build
2. To allow us to start a real voctocore in a container for integration tests later on.